discovery/addr.go

```diff
@@ -0,0 +1,83 @@
+package discovery
+
+import (
+	"fmt"
+	"net"
+	"sync"
+)
+
+type addrStatus bool
+
+const (
+	OK    = true
+	NotOK = false
+)
+
+type Addr struct {
+	// Something with IP + Port
+	net.TCPAddr
+}
+
+func MakeAddr(ipStr string, port int) (Addr, error) {
+	ip := net.ParseIP(ipStr)
+	if ip == nil {
+		return Addr{}, fmt.Errorf("unable to parse ip: %q", ip)
```

Comment:
Did you mean `ipStr` here? Otherwise, I think this will always return `unable to parse ip: <nil>`.

```suggestion
		return Addr{}, fmt.Errorf("unable to parse ip: %q", ipStr)
```